R/render.r

```diff
@@ -94,18 +94,43 @@ data_template <- function(pkg = ".", depth = 0L) {
 
 data_open_graph <- function(pkg = ".") {
   pkg <- as_pkgdown(pkg)
-  og <- list()
-  logo <- find_logo(pkg$src_path)
-  if (!is.null(logo)) {
+  og <- pkg$meta$template$opengraph %||% list()
+  og <- keep_supported_open_graph(og)
+  if (is.null(og$image) && !is.null(find_logo(pkg$src_path))) {
+    og$image <- list(src = path_file(find_logo(pkg$src_path)))
+  }
+  if (!is.null(og$image) && !grepl("^http", og$image$src)) {
     site_url <- pkg$meta$url %||% "/"
     if (!grepl("/$", site_url)) {
       site_url <- paste0(site_url, "/")
     }
-    og$image <- paste0(site_url, path_file(logo))
+    og$image$src <- gsub("^man/figures/", "reference/figures/", og$image$src)
+    og$image$src <- paste0(site_url, og$image$src)
+  }
+  if (length(og$twitter) &&
+      is.null(og$twitter$creator) &&
+      is.null(og$twitter$site)) {
+    abort(
+      "Twitter open graph information must include either 'creator' or 'site'."
+    )
   }
+  og$twitter$creator <- og$twitter$creator %||% og$twitter$site
+  og$twitter$site <- og$twitter$site %||% og$twitter$creator
   og
 }
 
+keep_supported_open_graph <- function(og) {
+  supported_fields <- c("image", "twitter")
+  unsupported_fields <- setdiff(names(og), supported_fields)
```

Comment:
Maybe you should also test that these are lists here?

Reply:
`check_open_graph()` now checks that `og`, `og$twitter`, `og$image` are all lists and provides informative error messages if a length-1 string is provided for either `twitter` or `image`.